libafl/src/mutators/grimoire.rs

```diff
@@ -281,7 +281,12 @@ where
                     continue;
                 }
                 let mut i = 0;
-                while i < bytes.len() - token_1.len() {
+                while bytes
+                    .len()
+                    .checked_sub(token_1.len())
+                    .map(|len| i < len)
+                    .unwrap_or(false)
+                {
```

Comment:
This check previously overflowed if a token was removed, but the bytes were subsequently shorter than the length of the token.